include/fmt/compile.h

```diff
@@ -36,7 +36,7 @@ template <typename OutputIt> class truncating_iterator_base {
   using difference_type = std::ptrdiff_t;
   using pointer = void;
   using reference = void;
-  using _Unchecked_type =
+  using Unchecked_type_ =
```

Comment:
This will introduce warnings in MSVC, please revert.

Reply:
What warning would it introduce?

Underscore followed by uppercase is reserved in C++: https://en.cppreference.com/w/cpp/language/identifiers

Reply:
Warnings about (un)checked iterators.

Reply:
Is `_Unchecked_type` some kind of Microsoft extension keyword?

Reply:
Sort of. It's not a keyword but some mechanism to opt out iterators.

Reply:
Could we somewhere do something like:

```c++
#if defined(_MSC_VER)
  #define fmt_unchecked_type _Unchecked_type
#else
  #define fmt_unchecked_type
#endif
```

Reply:
If you have access to MSVC 2015, I'd recommend checking if removing `_Unchecked_type` altogether still produces warnings. It might be no longer the case as IIRC they have been moving away from the checked iterators. You could even submit a test PR to the fmt repo and check warnings in appveyor: https://ci.appveyor.com/project/vitaut/fmt.

Reply:
Actually, we can see on this PR that there are definitely more warnings: https://ci.appveyor.com/project/vitaut/fmt/builds/42816914/job/briwktqjiiw9oa34. The macro approach seems OK but not that macros should be uppercase.

Reply:
OK, see new attempt.